code/game/machinery/colony_floodlights.dm

```diff
@@ -54,37 +54,37 @@
 		if(ispowered && turned_on)
 			toggle_lights()
 		ispowered = FALSE
-		turned_on = 0
+		turned_on = FALSE
 		update_icon()
 	else
 		ispowered = TRUE
 		update_icon()
 
 /obj/structure/machinery/colony_floodlight_switch/proc/toggle_lights()
-	for(var/obj/structure/machinery/colony_floodlight/F in floodlist)
-		spawn(rand(0,50))
-			F.is_lit = !F.is_lit
-			if(!F.damaged)
-				if(F.is_lit) //Shut it down
-					F.set_light(F.lum_value)
+	for(var/obj/structure/machinery/colony_floodlight/floodlight in floodlist)
+		spawn(rand(0, 50))
+			floodlight.is_lit = !floodlight.is_lit
+			if(!floodlight.damaged)
+				if(floodlight.is_lit) //Shut it down
+					floodlight.set_light(floodlight.lum_value)
 				else
-					F.set_light(0)
-			F.update_icon()
-	return 0
+					floodlight.set_light(0)
+			floodlight.update_icon()
+	return FALSE
```

Comment:
Github won't let me make an actual 'suggestion' here, but it might be worth changing this to an `addtimer()`:
```dm
/obj/structure/machinery/colony_floodlight_switch/proc/toggle_lights()
	for(var/obj/structure/machinery/colony_floodlight/floodlight as anything in floodlist)
		addtimer(CALLBACK(floodlight, TYPE_PROC_REF(/obj/structure/machinery/colony_floodlight, toggle_light)), rand(0, 5 SECONDS))
```
```dm
/obj/structure/machinery/colony_floodlight/proc/toggle_light()
	is_lit = !is_lit
	if(!damaged)
		if(is_lit)
			set_light(lum_value)
		else
			set_light(0)
	update_icon()
```